front/src/iframe_api.ts

```diff
@@ -73,8 +101,126 @@ class Popup {
         }, '*');
     }
 }
+function uuidv4() {
+    return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, (c) => {
+        const r = Math.random() * 16 | 0, v = c === 'x' ? r : (r & 0x3 | 0x8);
+        return v.toString(16);
+    });
+}
+function getGameState(): Promise<GameStateEvent> {
+    if (immutableData) {
+        return Promise.resolve(immutableData);
+    }
+    else {
+        return new Promise<GameStateEvent>((resolver, thrower) => {
+            gameStateResolver.push(resolver);
+            window.parent.postMessage({
+                type: "getState"
+            }, "*")
+        })
+    }
+}
+
+function getDataLayer(): Promise<DataLayerEvent> {
+    return new Promise<DataLayerEvent>((resolver, thrower) => {
+        dataLayerResolver.push(resolver);
+        postToParent({
+            type: "getDataLayer",
+            data: undefined
+        })
+    })
+}
+
+function getTag(): Promise<TagEvent> {
+    return new Promise<TagEvent>((resolver, thrower) => {
+        tagResolver.push((resolver));
+        postToParent({
+            type: "getTag",
+            data: undefined
+        })
+    })
+}
+
+const gameStateResolver: Array<(event: GameStateEvent) => void> = []
+const dataLayerResolver: Array<(event: DataLayerEvent) => void> = []
+const tagResolver: Array<(event : TagEvent) => void> = []
+let immutableData: GameStateEvent;
+
+const callbackPlayerMoved: { [type: string]: HasPlayerMovedEventCallback | ((arg?: HasPlayerMovedEvent | never) => void) } = {}
+
+
+function postToParent(content: IframeEvent<keyof IframeEventMap>) {
+    window.parent.postMessage(content, "*")
+}
+let playerUuid: string | undefined;
 
 window.WA = {
+
+    onPlayerMove(callback: HasPlayerMovedEventCallback): void {
+        playerUuid = uuidv4();
+        callbackPlayerMoved[playerUuid] = callback;
```

Comment:
Why do we need to playerUuid? Can't we simply add the callback to a simple list?

Reply:
i think that was sort of going in the direction of this 
https://github.com/thecodingmachine/workadventure/issues/1059
should be unneccessay in this commit though